src/software/guides/addin-storage.md

```diff
@@ -0,0 +1,384 @@
+
+```javascript
+api.call("Get",
+{
+  "typeName": "AddInData",
+  "search": {
+    "addInId": "a2C4ABQuLFkepPVf6-4OKAQ",
+    "selectClause": "items.[].name",
+    "whereClause": "customer.email = \"joe@smith.com\""
+  }
+});
+```
+This method call will return an array with multiple AddInData objects that satisfy both the select and where clauses.
+
+```json
+"result": [
+  {
+    "addInId": "a2C4ABQuLFkepPVf6-4OKAQ",
+    "id": "afLvRdUtXrE2D-XLwvqAgZQ",
+    "groups": [{"children": [], "id": "GroupCompanyId"}],
+    "details": "bottles"
+  },
+  {
+    "addInId": "a2C4ABQuLFkepPVf6-4OKAQ",
+    "id": "afLvRdUtXrE2D-XLwvqAgZQ",
+    "groups": [{"children": [], "id": "GroupCompanyId"}],
+    "details": "caps"
+  }
+]
+```
+**Note** : Both returned AddInData objects will have the same Id because they come from the same object in the database.
+
+**Example**  **4** : Get all data
+
+```javascript
+api.call("Get",
+{
+  "typeName": "AddInData",
+  "search": {
+    "addInId": "a2C4ABQuLFkepPVf6-4OKAQ"
+  }
+});
+```
+
+This method call will return an array with an AddInData object containing all the data stored in the AddInData object with the Id of a2C4ABQuLFkepPVf6-4OKAQ.
+
+```javascript
+"result": [
+  {
+    "addInId": "a2C4ABQuLFkepPVf6-4OKAQ",
+    "id": "afLvRdUtXrE2D-XLwvqAgZQ",
+    "groups": [{"children": [], "id": "GroupCompanyId"}],
+    "details":Object  { ... }
+        date: "2016-01-01T00:00:00.000Z"
+        -customer: Object
+            email: "joe@smith.com"
+            name: "joesmith"
+        -items: Array[2]
+            -0: Object
+                name: "bottles"
+                price: 12
+            -1: Object
+                name: "caps"
+                price: 20
+            length: 2
+  }
+]
+```
+### **Object Path Notation**
+
+The SELECT and WHERE clauses of the AddInDataSearch object use a special notation to describe an object path.
+
+If we wanted to modify the call in Example 4 to retrieve just the customer name from the AddInData object, we would add the following path notation to the SELECT clause of the AddInDataSearch object:
+
+**customer.name**
+
+```javascript
+api.call("Get",
+{
+  "typeName": "AddInData",
+  "search": {
+    "addInId": "a2C4ABQuLFkepPVf6-4OKAQ",
+    "selectClause": "customer.name"
+  }
+});
+```
+
+The returned AddInData object will contain a value of "joesmith" in itsdata property.
+
+If you have an array in the path, it must be indicated by a [] after the name of the array property.
+
+For example, if you wanted to modify Example 4 to select all item names, we would add the following to the SELECT clause of the AddInDataSearch object:
+
+**items.[].name**
+
+```javascript
+api.call("Get",
+{
+  "typeName": "AddInData",
+  "search": {
+    "addInId": "a2C4ABQuLFkepPVf6-4OKAQ",
+    "selectClause": "items.[].name"
+  }
+});
+```
+
+The same notation is used for theWHERE clause. This notation can be used to drill down to as many objects as you want.
```

Comment:
theWHERE => the WHERE

Reply:
Fixed in 764501adf2514edddd8a2468e0a9123abd659036